wgpu/src/triangle.rs

```diff
@@ -363,20 +387,17 @@ impl Pipeline {
                     &[(std::mem::size_of::<Uniforms>() * i) as u32],
                 );
 
-                render_pass.set_index_buffer(
-                    &self.index_buffer.raw,
-                    index_offset * std::mem::size_of::<u32>() as u64,
-                    0,
-                );
+                render_pass.set_index_buffer(self.index_buffer.raw.slice(..));
 
-                render_pass.set_vertex_buffer(
-                    0,
-                    &self.vertex_buffer.raw,
-                    vertex_offset * std::mem::size_of::<Vertex2D>() as u64,
-                    0,
-                );
+                render_pass
+                    .set_vertex_buffer(0, self.vertex_buffer.raw.slice(..));
 
-                render_pass.draw_indexed(0..indices as u32, 0, 0..1);
+                render_pass.draw_indexed(
+                    index_offset as u32
```

Comment:
perhaps it's better to pass down the offset in `index_buffer.raw.slice(index_offset * size_of::<u32>()..)`?

Reply:
I actually tried before resorting to this, but I forgot to multiply by `size_of` as usual and got a driver crash :sweat_smile: 

Changed in 7559e4fb30a282271717f700cb6e420cdf6123f0.